pkg/machinery/extensions/validate.go

```diff
@@ -12,28 +12,21 @@ import (
 	"strings"
 
 	"github.com/blang/semver/v4"
-
-	"github.com/siderolabs/talos/pkg/machinery/version"
 )
 
 // Validate the extension: compatibility, contents, etc.
-func (ext *Extension) Validate() error {
-	if err := ext.validateConstraints(); err != nil {
+func (ext *Extension) Validate(talosVersion semver.Version) error {
+	if err := ext.validateConstraints(talosVersion); err != nil {
 		return err
 	}
 
 	return ext.validateContents()
```

Comment:
how about `validateContents()` - as it's different for different versions of Talos, it might still break?

Reply:
I guess then  we need have something similar to quirks for extension allowed paths

Reply:
yep, either that, or a way to disable it in `Validate()` - e.g. Image Factory works on actually approved/shipped versions, so contents should be fine?